rasa/core/agent.py

```diff
@@ -289,6 +289,7 @@ def __init__(
         # Initializing variables with the passed parameters.
         self.domain = self._create_domain(domain)
```

Comment:
_create_domain also calls Domain.load(). Maybe we put the validation inside that method instead of calling the method in a separate `validate_domain()` function?